### PR TITLE
Enable the conversion API to work across allowed types.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: test/cpp/test_tensor.cpp [hunk 1]
```diff
@@ -1,18 +1,75 @@
...
 #include "torch/csrc/autograd/variable.h"
+#include "torch_xla/csrc/tensor.h"
+#include "torch_xla/csrc/tensor_util.h"
 #include "torch_xla_test.h"
 
 namespace torch_xla {
 namespace cpp_test {
+namespace {
+
+bool IsBidirectionalConversion(
```
Comment (asuhan) [inline test/cpp/test_tensor.cpp]: This name is confusing, I'd rather have a name which reflects that it's a test helper. How about `CheckBidirectionalConversion`?

File: torch_xla/csrc/tensor_util.cpp [hunk 2]
```diff
@@ -318,39 +356,30 @@ at::Tensor MakeTensorFromXlaLiteral(const xla::Literal& literal,
...
-    }
-    case xla::PrimitiveType::S32: {
-      return XlaLiteralToTensor<xla::int32, int32_t>(
-          literal, at::ScalarType::Int, xla::PrimitiveType::S32);
-    }
-    case xla::PrimitiveType::S64: {
-      return XlaLiteralToTensor<xla::int64, int64_t>(
-          literal, at::ScalarType::Long, xla::PrimitiveType::S64);
-    }
+    case xla::PrimitiveType::PRED:
```
Comment (asuhan) [inline torch_xla/csrc/tensor_util.cpp]: Do the tests pass on TPU? This is significantly more general than before.

File: test/cpp/test_tensor.cpp (hunk 1 above)
Comment (dlibenzi) [reply test/cpp/test_tensor.cpp]: SGTM

File: torch_xla/csrc/tensor_util.cpp (hunk 2 above)
Comment (dlibenzi) [reply torch_xla/csrc/tensor_util.cpp]: The test does not run on device. It is a simple low-level conversion matching test.
When we map ATEN ScalarType to XLA primitive type, our APIs will hand over the proper XLA type.

**Pull request merged**